### PR TITLE
changed order of advanced filter and subgroups

### PR DESCRIPTION
**Pull request by dsnyder-fairway**
oprdering is : Gender, Race/Ethnicity, English Learners, 504, IEP, Migrant Status, Economic Disadvantage.

next trying to reposition some of the subgroups

screen shot 2018-02-27 at 10 39 56 am
screen shot 2018-02-27 at 10 40 03 am

### CODE REVIEW

File: webapp/src/main/webapp/src/app/aggregate-report/aggregate-report-form.component.ts [hunk 1]
```diff
@@ -39,6 +39,12 @@ const notEmpty = properties => control => {
   return control.value.length ? null : { notEmpty: properties };
 };
 
+/**
+ * Used to determine the ordering of the subgroups section
+ * @type {string[]}
+ */
+const subgroupOrdering = [ "Gender", "Ethnicity", "LEP", "Section504", "IEP", "MigrantStatus", "EconomicDisadvantage", "StudentEnrolledGrade" ];
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/aggregate-report/aggregate-report-form.component.ts]: this should probably go in the `aggregate-report-option.service.ts` where we correct other sort orders. that way no matter where the dimensions are displayed they will have the right sort order

File: webapp/src/main/webapp/src/app/aggregate-report/aggregate-report-form.component.ts [hunk 2]
```diff
@@ -173,6 +179,10 @@ export class AggregateReportFormComponent {
     this.columnItems = this.columnOrderableItemProvider.toOrderableItems(this.settings.columnOrder);
 
     this.options = optionMapper.map(this.aggregateReportOptions);
+    this.options.dimensionTypes.sort((a, b) => {
+      return subgroupOrdering.indexOf(a.value) - subgroupOrdering.indexOf(b.value);
+    });
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/aggregate-report/aggregate-report-form.component.ts]: same comment - move to option service

File: webapp/src/main/webapp/src/app/aggregate-report/aggregate-report-summary.component.ts [hunk 3]
```diff
@@ -155,19 +155,13 @@ export class AggregateReportSummary {
     if (!equalSize(options.ethnicities, settings.ethnicities)) {
       filterRows.push({
         label: translate('aggregate-reports.form.field.ethnicity.label'),
-          values: orAll(options.ethnicities, settings.ethnicities, code => translate(`common.ethnicity.${code}`))
+        values: orAll(options.ethnicities, settings.ethnicities, code => translate(`common.ethnicity.${code}`))
```
Comment (dsnyder-fairway) [inline webapp/src/main/webapp/src/app/aggregate-report/aggregate-report-summary.component.ts]: changes around this area were mostly unnecessary. Just thought I should maintain consistent ordering everywhere (in our code and on presentation)

File: webapp/src/main/webapp/src/app/aggregate-report/aggregate-report-form.component.ts [hunk 4]
```diff
@@ -39,6 +39,12 @@ const notEmpty = properties => control => {
   return control.value.length ? null : { notEmpty: properties };
 };
 
+/**
+ * Used to determine the ordering of the subgroups section
+ * @type {string[]}
```
Comment (malaffoon) [inline webapp/src/main/webapp/src/app/aggregate-report/aggregate-report-form.component.ts]: maybe include a code comment like "This should be coordinated with the order of the Advanced Filters shown on the same page, see aggregate-report-form.component.html.

**Comment by dsnyder-fairway** [comment]
@esotrope this works 👍

**Pull request merged**